validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/LocalValidatorSource.java

```diff
@@ -74,13 +84,59 @@ public LocalValidatorSource(
 
   @Override
   public boolean canAddValidator() {
-    return false;
+    return !readOnly;
   }
 
   @Override
-  public MutableValidatorAddResult addValidator(
+  public AddLocalValidatorResult addValidator(
       final KeyStoreData keyStoreData, final String password) {
-    throw new UnsupportedOperationException();
+    if (readOnly || maybeDataDirLayout.isEmpty()) {
```

Comment:
I'd be tempted to include the `maybeDataDirLayout.isEmpty` check in `canAddValidator` and then just make this `if (!canAddValidator())` so the two checks are always completely consistent.